src/main/java/com/github/shyiko/mysql/binlog/json/JsonBinary.java

```diff
@@ -0,0 +1,1002 @@
+     */
+    protected void parseInt32(JsonFormatter formatter) throws IOException {
+        int value = readInt32();
+        formatter.value(value);
+    }
+
+    /**
+     * Parse a 4 byte unsigned integer value.
+     *
+     * @param formatter the formatter to be notified of the parsed value; may not be null
+     * @throws IOException if there is a problem reading the JSON value
+     */
+    protected void parseUInt32(JsonFormatter formatter) throws IOException {
+        long value = readUInt32();
+        formatter.value(value);
+    }
+
+    /**
+     * Parse a 8 byte integer value.
+     *
+     * @param formatter the formatter to be notified of the parsed value; may not be null
+     * @throws IOException if there is a problem reading the JSON value
+     */
+    protected void parseInt64(JsonFormatter formatter) throws IOException {
+        long value = readInt64();
+        formatter.value(value);
+    }
+
+    /**
+     * Parse a 8 byte unsigned integer value.
+     *
+     * @param formatter the formatter to be notified of the parsed value; may not be null
+     * @throws IOException if there is a problem reading the JSON value
+     */
+    protected void parseUInt64(JsonFormatter formatter) throws IOException {
+        BigInteger value = readUInt64();
+        formatter.value(value);
+    }
+
+    /**
+     * Parse a 8 byte double value.
+     *
+     * @param formatter the formatter to be notified of the parsed value; may not be null
+     * @throws IOException if there is a problem reading the JSON value
+     */
+    protected void parseDouble(JsonFormatter formatter) throws IOException {
+        long rawValue = readInt64();
+        double value = Double.longBitsToDouble(rawValue);
+        formatter.value(value);
+    }
+
+    /**
+     * Parse the length and value of a string stored in MySQL's "utf8mb" character set (which equates to Java's
+     * {@link StandardCharsets#UTF_8} character set. The length is a {@link #readVariableInt(ByteBuffer)
+     * variable length integer} length of the string.
+     *
+     * @param offset the offset at which the value is to be read; may not be null
+     * @param formatter the formatter to be notified of the parsed value; may not be null
+     * @throws IOException if there is a problem reading the JSON value
+     */
+    protected void parseString(JsonFormatter formatter) throws IOException {
+        int length = readVariableInt();
+        byte[] content = new byte[length];
+        reader.read(content);
+        String value = new String(content, UTF_8);
+        formatter.value(value);
+    }
+
+    /**
+     * Parse an opaque type. Specific types such as {@link #parseDate(Offset, formatter) DATE},
+     * {@link #parseTime(Offset, formatter) TIME},{@link #parseDatetime(Offset, formatter) DATETIME}, and
+     * {@link #parseTimestamp(Offset, formatter) TIMESTAMP} values are stored as opaque types, though they are
```

Comment:
JavaDoc mentions ColumnType.TIMESTAMP but it seems like it's not handled (JsonFormatter.valueTimestamp is also not used). Is this intentional?